internal/storage/pmem/pmem.go

```diff
@@ -44,69 +50,168 @@ func mountInternal(source, target string) (err error) {
 	return nil
 }
 
-// Mount mounts the pmem device at `/dev/pmem<device>` to `target`.
+// Mount mounts the pmem device at `/dev/pmem<device>` to `target` in a basic scenario.
+// In the case when mappingInfo or/and verityInfo is/are not nil, device-mapper framework
+// is used to create linear and verity targets accordingly.
```

Comment:
If either mappingInfo or verifyInfo are non-nil, the device-mapper framework is used to create linear and verify targets accordingly. If both are non-nil, the linear target is created first and used as the data/hash device for the verity target.

Reply:
I'm not sure where this code explicitly causes the linear target to be used as the hash device for the verity target. Did I miss it or is this relationship actually established by the caller of Mount?

Reply:
it's [this] (https://github.com/microsoft/opengcs/blob/c1027f7c2c8b7b30dc38e204deaf9df773258a3a/internal/storage/pmem/pmem.go#L81-L107) portion of code that does it. Note that I'm reassigning `devicePath` variable, rather than creating a local one. now that I think about it, I should probably do the reassignment explicitly, since the code will be more clear.

Reply:
Oh I missed that they were related by passing in the devicePath. Sure explicit reassignment might make it clearer though it probably isn't necessary.